packages/compat/src/fixup-rules.js

```diff
@@ -98,52 +138,158 @@ export function fixupRule(ruleDefinition) {
+					return sourceCode.getTokenBefore(node, {
+						includeComments: true,
+						skip,
+					});
+				},
+				getTokenOrCommentAfter(node, skip) {
+					return sourceCode.getTokenAfter(node, {
+						includeComments: true,
+						skip,
+					});
+				},
+				isSpaceBetweenTokens(first, second) {
+					return sourceCode.isSpaceBetween(first, second);
+				},
+				getJSDocComment(node) {
+					let parent = node.parent;
+
+					switch (node.type) {
+						case "ClassDeclaration":
+						case "FunctionDeclaration":
+							return findJSDocComment(
+								looksLikeExport(parent) ? parent : node,
+								sourceCode,
+							);
+
+						case "ClassExpression":
+							return findJSDocComment(parent.parent, sourceCode);
+
+						case "ArrowFunctionExpression":
+						case "FunctionExpression":
+							if (
+								parent.type !== "CallExpression" &&
+								parent.type !== "NewExpression"
+							) {
+								while (
+									!sourceCode.getCommentsBefore(parent)
+										.length &&
+									!/Function/u.test(parent.type) &&
+									parent.type !== "MethodDefinition" &&
+									parent.type !== "Property"
+								) {
+									parent = parent.parent;
+
+									if (!parent) {
+										break;
+									}
+								}
+
+								if (
+									parent &&
+									parent.type !== "FunctionDeclaration" &&
+									parent.type !== "Program"
+								) {
+									return findJSDocComment(parent, sourceCode);
+								}
+							}
+
+							return findJSDocComment(node, sourceCode);
+
+						default:
+							return null;
+					}
+				},
+			});
+
+			Object.freeze(compatSourceCode);
+		}
+
+		let currentNode = compatSourceCode.ast;
 
-		const newContext = Object.assign(Object.create(context), {
-			parserServices: sourceCode.parserServices,
+		const compatContext = Object.assign(Object.create(context), {
+			parserServices: compatSourceCode.parserServices,
 
 			/*
 			 * The following methods rely on the current node in the traversal,
 			 * so we need to add them manually.
 			 */
 			getScope() {
-				return sourceCode.getScope(currentNode);
+				return compatSourceCode.getScope(currentNode);
 			},
 
 			getAncestors() {
-				return sourceCode.getAncestors(currentNode);
+				return compatSourceCode.getAncestors(currentNode);
 			},
 
 			markVariableAsUsed(variable) {
-				sourceCode.markVariableAsUsed(variable, currentNode);
+				compatSourceCode.markVariableAsUsed(variable, currentNode);
 			},
 		});
 
+		if (eslintVersion >= 10) {
+			Object.assign(compatContext, {
+				parserOptions: compatContext.languageOptions.parserOptions,
+
+				getCwd() {
+					return compatContext.cwd;
+				},
+
+				getFilename() {
+					return compatContext.filename;
+				},
+
+				getPhysicalFilename() {
+					return compatContext.physicalFilename;
+				},
+
+				getSourceCode() {
+					return compatSourceCode;
+				},
+			});
+
+			Object.defineProperty(compatContext, "sourceCode", {
+				get() {
+					return compatSourceCode;
+				},
```

Comment:
```suggestion
				enumerable: true,
				value: compatSourceCode,
```

For compatibility, it might be better to make it be like the original property.